src/Shipping/ShippingController.php

```diff
@@ -1,24 +1,180 @@
 		add_filter( 'woocommerce_local_pickup_methods', array( $this, 'filter_local_pickup_methods' ) );
 		add_filter( 'woocommerce_customer_taxable_address', array( $this, 'pickup_location_customer_tax_location' ) );
 	}
 
+	/**
+	 * Register Local Pickup settings for rest api.
+	 */
+	public function register_settings() {
+		register_setting(
+			'options',
+			'woocommerce_pickup_location_settings',
+			[
+				'type'         => 'object',
+				'description'  => 'WooCommerce Local Pickup Method Settings',
+				'default'      => [],
+				'show_in_rest' => [
+					'name'   => 'pickup_location_settings',
+					'schema' => [
+						'type'       => 'object',
+						'properties' => array(
+							'enabled'    => [
+								'description' => __( 'If enabled, this method will appear on the block based checkout.', 'woo-gutenberg-products-block' ),
+								'type'        => 'string',
+								'enum'        => [ 'yes', 'no' ],
+							],
+							'title'      => [
+								'description' => __( 'This controls the title which the user sees during checkout.', 'woo-gutenberg-products-block' ),
+								'type'        => 'string',
+							],
+							'tax_status' => [
+								'description' => __( 'If a cost is defined, this controls if taxes are applied to that cost.', 'woo-gutenberg-products-block' ),
+								'type'        => 'string',
+								'enum'        => [ 'taxable', 'none' ],
+							],
+							'cost'       => [
+								'description' => __( 'Optional cost to charge for local pickup.', 'woo-gutenberg-products-block' ),
+								'type'        => 'string',
+							],
+						),
+					],
+				],
+			]
+		);
+		register_setting(
+			'options',
+			'pickup_location_pickup_locations',
+			[
+				'type'         => 'array',
+				'description'  => 'WooCommerce Local Pickup Locations',
+				'default'      => [],
+				'show_in_rest' => [
+					'name'   => 'pickup_locations',
+					'schema' => [
+						'type'  => 'array',
+						'items' => [
+							'type'       => 'object',
+							'properties' => array(
+								'name'    => [
+									'type' => 'string',
+								],
+								'address' => [
+									'type'       => 'object',
+									'properties' => array(
+										'address_1' => [
+											'type' => 'string',
+										],
+										'city'      => [
+											'type' => 'string',
+										],
+										'state'     => [
+											'type' => 'string',
+										],
+										'postcode'  => [
+											'type' => 'string',
+										],
+										'country'   => [
+											'type' => 'string',
+										],
+									),
+								],
+								'details' => [
+									'type' => 'string',
+								],
+								'enabled' => [
+									'type' => 'boolean',
+								],
+							),
+						],
+					],
+				],
+			]
+		);
+	}
+
+	/**
+	 * Load admin scripts.
+	 */
+	public function admin_scripts() {
+		$this->asset_api->register_script( 'wc-shipping-method-pickup-location', 'build/wc-shipping-method-pickup-location.js', [], false );
```

Comment:
This script does have translation, the last one should be true not false.

Reply:
https://github.com/woocommerce/woocommerce-blocks/pull/7581/commits/a07560031f013e78109c3551bc251db19ac4215f